test/e2e-tests-upgrade.sh

```diff
@@ -24,6 +24,10 @@
 # Scenario 2: install the previous release, create the pipelines and tasks, upgrade
 # to the current release, and validate whether the Tekton pipeline works.
 
+# Scenario 3: install the current pipelines, use the integration tests from the previous release(s)
+# to prevent regression. Incompatible changes made in the current release cycle are expected to fail
+# and will be skipped after failing once.
```

Comment:
How does the skipping work? Is there a way to mark as test as skipped for the upgrade test suite?

Reply:
- We could introduce a new build tag for upgrade tests, but this would require updates to the e2e test tag when a feature is introduced or there are any breaking changes made to it and spotted by the upgrade test.
- We could have a 'skip list'(probably in the e2e-test-upgrade.sh) and use regex and `mv` command to move around skipped tests in the `./test` folder.

I thought this could be in a followup PR? if not so happy to add in this PR if any of the above sounds ok to add

Reply:
Oh, I forgot upgrade tests are a cron job. I wonder if it would make sense to run them as part of CI instead? We might be constrained by our test cluster bandwidth though just based on our experience with integration tests.

Of the two approaches you mentioned, I think the "skip list" is better because it's opt-out rather than opt-in.

Reply:
Thanks @lbernick , I do agree that it would be nice and responsive to detect breaking changes in CI. Added https://github.com/tektoncd/pipeline/issues/6709 as a followup to track moving upgrade tests to CI.

I have also tried out the "skip list" approach in the current commit.

Reply:
> Oh, I forgot upgrade tests are a cron job. I wonder if it would make sense to run them as part of CI instead? We might be constrained by our test cluster bandwidth though just based on our experience with integration tests.

One idea would be to run these tests as part of the release pipeline

Reply:
Thanks Dibyo, tracked the upcoming goals at https://github.com/tektoncd/pipeline/issues/6709